### ISSUE TITLE
(bug) [Mat-table]: sorting indicator not visible under high contrast mode

### ISSUE THREAD

**Issue by shb98** [angular/components#24139]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

mat-table sort indicator not visible under high contrast mode

### Reproduction

Steps to reproduce:
1. Switch to Windows high contrast mode 
![image](https://user-images.githubusercontent.com/10054756/147512418-b2f5ae34-06fa-419a-a528-4e740266c185.png)

2. Go to the official [sample mat-table sorting page](https://material.angular.io/components/table/overview#sorting)
3. Click the header to sort

### Expected Behavior

Sort indicator (up/down arrow) should be visible under high contrast mode as normal display mode

### Actual Behavior

Sort indicator (up/down arrow) is NOT visible under high contrast mode as below
![image](https://user-images.githubusercontent.com/10054756/147512566-d3081273-ece2-41e4-97aa-c1f68ed3a26c.png)

### Environment

- Angular: 12
- CDK/Material: 12
- Browser(s): ANY
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

**Comment by crisbeto** [comment]
It works for me on Windows 10. Did you refresh the page after enabling high contrast mode? We only check for it when the app is initialized.
![Table__Angular_Material_-_Google_Chrome_2021-12-28_10-38-51](https://user-images.githubusercontent.com/4450522/147546872-9cc3398d-2f8a-4d18-8da4-20bc5fb6ad3a.png)

**Comment by shb98** [comment]
> It works for me on Windows 10. Did you refresh the page after enabling high contrast mode? We only check for it when the app is initialized. ![Table__Angular_Material_-_Google_Chrome_2021-12-28_10-38-51](https://user-images.githubusercontent.com/4450522/147546872-9cc3398d-2f8a-4d18-8da4-20bc5fb6ad3a.png)

Yes, I force refreshed the page, and also tried restarting browser in private mode. My OS is Windows 11. I can repro the issue on both Edge and Chrome. 

![image](https://user-images.githubusercontent.com/10054756/147961247-7776d908-f813-4ab5-afdb-7963d0dd88a5.png)

**Comment by shb98** [comment]
Looks the CDK [high-contrast-mode-detector](https://github.com/angular/components/blob/master/src/cdk/a11y/high-contrast-mode/high-contrast-mode-detector.ts) could not handle all the high contrast themes well as it assumed to be absolute 'white/black' or absolute 'black/white'. 

There is another way to check the high contrast activeness - https://www.scottohara.me/note/2021/10/01/detect-high-contrast-and-dark-modes.html

**Comment by crisbeto** [comment]
The reason we're doing it this way is to better support browsers which don't support the media queries. It's still a valid a issue that we don't detect the other themes though.

**Comment by jelbourn** [comment]
@crisbeto we do need to start supporting `forced-colors` soon, so it might be a good time to adjust how we do high contrast detection (especially with IE11 gone)

**Comment by crisbeto** [comment]
Switching it to `force-colors` is straightforward, but we won't be able to distinguish between white-on-black and black-on-white contrast modes. I don't think that we have any CSS that actually distinguishes between the two either so we could just switch back to using media queries and deprecate the detector.

**Comment by jelbourn** [comment]
I wonder how widely anyone uses the different values. My hunch is that we can probably remove `HighContrastModeDetector` as a dependency of our components and let it keep chugging along as a standalone thing.

**Comment by shb98** [comment]
> Switching it to `force-colors` is straightforward, but we won't be able to distinguish between white-on-black and black-on-white contrast modes. I don't think that we have any CSS that actually distinguishes between the two either so we could just switch back to using media queries and deprecate the detector.

if using force-colors, you don't really need to distinguish white-on-black and black-on-white? Just follow and use the forced color should be good? As my case only need to support browser that support force-colors media query, adding these override works for my case. 

```CSS
/* High contrast modes */
@media (forced-colors: active) {
  /* mat-table sorting indicator accessbility issue workaround */
  .mat-sort-header-pointer-left,
  .mat-sort-header-pointer-right {
    width: 0 !important;
    height: 0 !important;
    border-left: solid 6px;
    border-top: solid 2px;
    border-color: windowText;
  }

  .mat-sort-header-pointer-middle {
    width: 0 !important;
    height: 0 !important;
    border-top: solid 2px;
    border-left: solid 2px;
    border-color: windowText;
  }

  .mat-sort-header-stem {
    width: 0 !important;
    border-left: solid 2px;
    border-color: windowText;
  }
}
```

**Comment by crisbeto** [comment]
This should be resolved by https://github.com/angular/components/pull/25624.

**Comment by Totati** [comment]
Yes it was.
Hover:
![image](https://user-images.githubusercontent.com/17689719/196358742-772595b8-9d56-4eb2-8f95-73ac73df1c07.png)
Active:
![image](https://user-images.githubusercontent.com/17689719/196358831-ae1d3ccd-8d2f-4a6d-83cb-d650f859acff.png)
